src/builder.rs

```diff
@@ -0,0 +1,141 @@
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+
+use anyhow::Result;
+use hyper::{Body, Request, Response};
+
+use crate::app_context::AppContext;
+use crate::async_graphql_hyper::{GraphQLBatchRequest, GraphQLRequest};
+use crate::blueprint::{validate_operations, Blueprint, OperationQuery};
+use crate::config::{Config, ConfigModule, ConfigReader, Source};
+use crate::fmt::Fmt;
+use crate::http::handle_request;
+use crate::print_schema;
+use crate::runtime::TargetRuntime;
+use crate::valid::Validator;
+
+/// Struct to build TailcallExecutor
+#[derive(Clone)]
+pub struct TailcallBuilder {
+    /// Holds a list of file paths
+    files: Vec<String>,
+    /// Holds a list of schema information
+    schemas: Vec<SchemaHolder>,
+}
+
+/// Struct to hold schema information while building executor
+#[derive(Clone)]
+struct SchemaHolder {
+    /// Holds a type of schema
+    source: Source,
+    /// Holds content of schema
+    schema: String,
+    /// Holds a path to parent dir for content in @link
+    parent_dir: Option<PathBuf>,
+}
+
+#[derive(Clone)]
+pub struct TailcallExecutor {
+    pub config_module: ConfigModule,
+    pub app_ctx: Arc<AppContext>,
+}
+
+impl TailcallBuilder {
+    /// Creates a new instance of TailcallBuilder
+    pub fn new() -> Self {
+        Self { files: vec![], schemas: vec![] }
+    }
+
+    /// Adds a configuration source with schema and optional parent directory
+    pub fn with_config_source<T: ToString, P: AsRef<Path>>(
+        mut self,
+        source: Source,
+        schema: T,
+        parent_dir: Option<P>,
+    ) -> Self {
+        let parent_dir = parent_dir.map(|p| p.as_ref().into());
+        self.schemas
+            .push(SchemaHolder { source, schema: schema.to_string(), parent_dir });
+        self
+    }
+
+    /// Adds an array of paths to configuration files
+    /// The file IO is carried out using runtime passed in build function
+    pub fn with_config_files<T: ToString>(mut self, files: &[T]) -> Self {
+        self.files
+            .push(files.iter().map(|v| v.to_string()).collect());
+        self
+    }
+
+    /// Builds TailcallExecutor with the provided runtime
+    pub async fn build(self, runtime: TargetRuntime) -> Result<TailcallExecutor> {
```

Comment:
```rust
pub async fn build_with(self, runtime: FnOnce(Blueprint) -> Result<TargetRuntime>) -> Result<TailcallExecutor>
```
Add a new method to customize the runtime based on the blueprint.

Reply:
Updated the implementation, now we do not need to update runtime in executor.